cmd/influxd/run/server_test.go

```diff
@@ -3169,6 +3169,53 @@ func TestServer_Query_GroupByTimeCutoffs(t *testing.T) {
 	}
 }
 
+// Test for bug #4481 GROUP BY on intervals < 1ms makes too many buckets
+func TestServer_Query_GroupBySmallInterval(t *testing.T) {
+	t.Parallel()
+	s := OpenServer(NewConfig(), "")
+	defer s.Close()
+
+	if err := s.CreateDatabaseAndRetentionPolicy("db0", newRetentionPolicyInfo("rp0", 1, 0)); err != nil {
+		t.Fatal(err)
+	}
+	if err := s.MetaStore.SetDefaultRetentionPolicy("db0", "rp0"); err != nil {
+		t.Fatal(err)
+	}
+
+	writes := []string{
+		fmt.Sprintf(`cpu value=1i %d`, time.Now().UnixNano()),
+	}
+	test := NewTest("db0", "rp0")
+	test.write = strings.Join(writes, "\n")
+
+	test.addQueries([]*Query{
+		&Query{
+			name:    "small interval - bug #4481",
+			params:  url.Values{"db": []string{"db0"}},
+			command: `SELECT MEAN(value) FROM cpu where time >= now() - 1u group by time(1u)`,
+			exp:     `"values":\[\["\d\d\d\d-\d\d-\d\dT\d\d:\d\d:\d\d(.\d*)?Z",null\],\["\d\d\d\d-\d\d-\d\dT\d\d:\d\d:\d\d(.\d*)?Z",null\]\]`,
```

Comment:
What are you using a regex? If it's because the time is not deterministic you should instead grab the value of now() and store that in a variable. Then you can use that variable in your write and form the expected string here.

Make sense? Doing this will make the test easier to understand.